### PR TITLE
AsyncStorage

### PR DESCRIPTION
**Pull request by david8611**
This implementation of a local client-side storage using Async Storage seems to be working. Since we have decided to use a functional component approach (https://medium.com/wesionary-team/react-functional-components-vs-class-components-86a2d2821a22) I decided to utilize the hook ```useEffect``` which will execute each time the current component renders and the other prop variable state changes. This hook can from my understanding be used as a substitute for the ```componentDidMount``` and ```componentDidUpdate``` when side effects need to be invoked as mensioned here: https://reactjs.org/docs/hooks-effect.html. It also seems to work well in unison with the ```useState``` hook. 
Remember to fetch the new dependency using ```npm install```. IOS users may or may not need to run ```npx pod-install``` as mensioned here: https://react-native-async-storage.github.io/async-storage/docs/install. 

Basically, to check if the stored settings work correctly. Try launching the tracking-settings view and change some settings. Then exit the app and run the project again on the same phone/device. The settings should then be stored.
To apply these Tracking settings in the Tracking view an import of the Async Storage is needed. The key-values should be the same and stored locally and should be available when selecting which elements/parts of the tracking view should be visible/active depending on the selected options in tracking-settings. Maybe @Kotune and @senko-tail could look into this?

To clear the Async Storage one could for example execute ```AsyncStorage.clear();``` in an async-function. This should erase as stored values.

### CODE REVIEW

**Comment by epulls** [comment]
Works for both my Samsung and in web. Although I noticed (I think) a probably unrelated visual bug again.
bild
The buttons are not centered on the screen. I'm very well aware of that these buttons don't serve any purpose for the finished product but it might still be something to look into. I've never seen the view so 
compressed.

I will give my approval but suggest that we wait for someone to test with iOS aswell.

**Review by epulls — APPROVED** [review]
Aside from the most likely unrelated visual bug, everythings looks and works as expected.

**Comment by SamuelHarner** [comment]
Settings being stored worked for me in iOS simulator as described in the above comment: https://github.com/ovulai/ovulai-react-native-app/pull/45#issue-583409691

I did not have to run `npx pod-install` for it to work.

**Review by SamuelHarner — APPROVED** [review]

**Comment by david8611** [comment]
> Works for both my Samsung and in web. Although I noticed (I think) a probably unrelated visual bug again.
> bild
> The buttons are not centered on the screen. I'm very well aware of that these buttons don't serve any purpose for the finished product but it might still be something to look into. I've never seen the view so
> compressed.
> 
> I will give my approval but suggest that we wait for someone to test with iOS aswell.

I think it is fine if se adress and fix these in https://github.com/ovulai/ovulai-react-native-app/pull/43
